pkg/runtime/compstore/components.go

```diff
@@ -90,6 +91,18 @@ func (c *ComponentStore) ListComponents() []compsv1alpha1.Component {
 	return comps
 }
 
+func (c *ComponentStore) ListMatchComponents(match string) []compsv1alpha1.Component {
```

Comment:
I would recommend changing this to prefix-matching. Then you can pass as prefix `middleware.`. This is safer than doing a match in the entire string, as imagine one day we had a component called `state.mymiddleware` for example :)

Reply:
Ya good catch tho